src/main/java/com/google/devtools/build/lib/remote/disk/DiskAndRemoteCacheClient.java

```diff
@@ -103,18 +103,22 @@ public void close() {
   @Override
   public ListenableFuture<ImmutableSet<Digest>> findMissingDigests(
       RemoteActionExecutionContext context, Iterable<Digest> digests) {
-    ListenableFuture<ImmutableSet<Digest>> remoteQuery =
-        remoteCache.findMissingDigests(context, digests);
     ListenableFuture<ImmutableSet<Digest>> diskQuery =
         diskCache.findMissingDigests(context, digests);
-    return Futures.whenAllSucceed(remoteQuery, diskQuery)
-        .call(
-            () ->
-                ImmutableSet.<Digest>builder()
-                    .addAll(remoteQuery.get())
-                    .addAll(diskQuery.get())
-                    .build(),
-            MoreExecutors.directExecutor());
+    if (!options.incompatibleRemoteResultsIgnoreDisk || options.remoteUploadLocalResults) {
```

Comment:
Can we use the helper method `shouldUploadLocalResultsToRemoteCache` here?

Reply:
Sure. I was copying the same check that was used in the method above.